tcmur_cmd_handler.c

```diff
@@ -2267,6 +2279,22 @@ static int tcmur_cmd_handler(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 		break;
 	}
 
+	/* setup timer for async handled commands */
+	if (!rhandler->has_timer) {
+		timer = malloc(sizeof(*timer));
+		if (!timer) {
+			tcmu_dev_err(dev, "malloc timer failed!\n");
+			goto untrack;
+		}
+
+		timer->data = cmd;
+		timer->expires = tcmu_get_dev_cmd_time_out(dev) + 5;
```

Comment:
Let's go ahead and just add a tcmu-runner specific timer just in case users will need this set differently from the kernel one. We might have where they are doing tape, and they cannot enable the kernel timer, but they want the userspace one, because we have more flexibility and can one day cancel/abort/cleanup commands on the physical backend.

We should be able to add this into the existing dev_config string with no kernel changes.

For rtslib UserBackedStorageObject:__init__() we might want to add a new arg for these type arguments for the daemon.

_configure() can then add these new args before the device ones. Something like:

tcmu-runner_args=cmd_timeout=$N;setting123=$ABC;setting345=$DEF/handler_args=what_is_getting_pased_to_the_handler_today

libtcmu.c_dev_added can then just chop its args out and store them in tcmu_device. tcmu_get/set_dev_cmd_time_out then just interact with the tcmu_device.

Reply:
Actually, I am a dummy :) There is no need to add the extra init() arg and make daemon args and all that extra stuff.

Just add it onto the normal hander args, have the handle parse it and call tcmu_set_dev_cmd_time_out() to set the value.

We can then drop the has_timer field.

Reply:
We should not make it tcmu.conf based because different devices may need different values.

Just pass it in with the other handler info on the cfg string.

Reply:
Okay.